tests/sentry/web/frontend/test_organization_auth_settings.py

```diff
@@ -554,3 +556,82 @@ def test_edit_sso_settings__scim(self):
                 scim_role_restricted_user.flags["idp:role-restricted"],
             )
         )
+
+
+dummy_provider_config = {
+    "idp": {
+        "entity_id": "https://example.com/saml/metadata/1234",
+        "x509cert": "foo_x509_cert",
+        "sso_url": "http://example.com/sso_url",
+        "slo_url": "http://example.com/slo_url",
+    },
+    "attribute_mapping": {
+        Attributes.IDENTIFIER: "user_id",
+        Attributes.USER_EMAIL: "email",
+        Attributes.FIRST_NAME: "first_name",
+        Attributes.LAST_NAME: "last_name",
+    },
+}
+
+
+class DummySAML2Provider(GenericSAML2Provider):
+    name = "dummy"
+
+    def get_saml_setup_pipeline(self):
+        return []
+
+    def build_config(self, state):
+        return dummy_provider_config
+
+
+@region_silo_test(stable=True)
+class OrganizationAuthSettingsSAML2Test(AuthProviderTestCase):
+    provider = DummySAML2Provider
+    provider_name = "saml2_dummy"
+
+    def setUp(self):
+        super().setUp()
+        self.user = self.create_user("foobar@sentry.io")
+        self.organization = self.create_organization(owner=self.user, name="saml2-org")
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            self.auth_provider = AuthProvider.objects.create(
+                provider=self.provider_name,
+                config=dummy_provider_config,
+                organization_id=self.organization.id,
+            )
+
+    def test_update_generic_saml2_config(self):
+        self.login_as(self.user, organization_id=self.organization.id)
+
+        expected_provider_config = {
+            "idp": {
+                "entity_id": "https://foobar.com/saml/metadata/4321",
+                "x509cert": "bar_x509_cert",
+                "sso_url": "http://foobar.com/sso_url",
+                "slo_url": "http://foobar.com/slo_url",
+            },
+            "attribute_mapping": {
+                Attributes.IDENTIFIER: "new_user_id",
+                Attributes.USER_EMAIL: "new_email",
+                Attributes.FIRST_NAME: "new_first_name",
+                Attributes.LAST_NAME: "new_last_name",
+            },
+        }
+
+        configure_path = reverse(
+            "sentry-organization-auth-provider-settings", args=[self.organization.slug]
+        )
+        payload = {
+            **expected_provider_config["idp"],
+            **expected_provider_config["attribute_mapping"],
+        }
+        resp = self.client.post(configure_path, payload)
+        assert resp.status_code == 200
+
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            actual = AuthProvider.objects.get(id=self.auth_provider.id)
+            assert actual.config == expected_provider_config
+            assert actual.config != self.auth_provider.config
+
+            assert actual.provider == self.auth_provider.provider
+            assert actual.flags == self.auth_provider.flags
```

Comment:
Should we add tests for cases in which this fails?

Reply:
This asserts that the flags remain unchanged. I haven't seen a code path that updates the flags for the generic SAML2 update flow.